File: Microsoft/Q1.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
class Solution{

  public:
	int minDifference(int arr[], int n)  { 
	    // Your code goes here
	    int sum = 0;
	    for(int i=0; i<n; i++) sum+=arr[i];
	    
	    int dp[n+1][(sum/2)+1];
	    
	    for(int i=0; i<=n; i++){
	        for(int j=0; j<=sum/2; j++){
	            if(j==0) dp[i][j] = 0;
	            else if(i==0) dp[i][j] = j;
	            else if(arr[i-1]>j) dp[i][j] = dp[i-1][j];
	            else dp[i][j] = min(dp[i-1][j], dp[i-1][j-arr[i-1]]);
	        }
	    }
	    
	    return (sum%2) + 2*dp[n][sum/2];
	} 
};


// { Driver Code Starts.
int main() 
{
   
   
   	int t;
    cin >> t;
    while (t--)
    {
        int n;
        cin >> n;

        int a[n];
        for(int i = 0; i < n; i++)
        	cin >> a[i];

       

	    Solution ob;
	    cout << ob.minDifference(a, n) << "\n";
	     
    }
    return 0;
}  // } Driver Code Ends

File: Adobe/Q1.py
class Solution:
    def subArraySum(self,arr, n, s): 
       #Write your code here
        i=0
        j=1
        lst=[]
        if arr[0]==s:
            lst=[1,1]
            return lst
        sm=arr[i]
        
        while j<n:
            sm+=arr[j]
            if sm==s:
                lst.append(i+1)
                lst.append(j+1)
                return lst
            if sm>s:
                while sm>s:
                    sm-=arr[i]
                    i+=1
                    if sm==s:
                        lst=[i+1,j+1]
                        return lst
            j+=1
        lst.append(-1)
        return lst
        
#{ 
#  Driver Code Starts
#Initial Template for Python 3

import math

def main():
        T=int(input())
        while(T>0):
            
            NS=input().strip().split()
            N=int(NS[0])
            S=int(NS[1])
            
            A=list(map(int,input().split()))
            ob=Solution()
            ans=ob.subArraySum(A, N, S)
            
            for i in ans:
                print(i, end=" ")
                
            print()
            
            T-=1


if __name__ == "__main__":
    main()
# } Driver Code Ends
